File: packages/network_analyzer_android/android/src/main/kotlin/com/koaladevelopments/network_analyzer_android/monitoring/MonitorErrors.kt
package com.koaladevelopments.network_analyzer_android.monitoring

/**
 * The error codes the native side raises across the channel.
 *
 * The Dart platform implementation maps each to a typed `Failure`, so no
 * `PlatformException` ever reaches a host application (constitution,
 * Principle III). Kept in one place because both platforms must agree.
 */
object MonitorErrors {
    const val PERMISSION_DENIED = "PERMISSION_DENIED"
    const val UNSUPPORTED_CAPABILITY = "UNSUPPORTED_CAPABILITY"
    const val GATEWAY_DISCOVERY_FAILED = "GATEWAY_DISCOVERY_FAILED"
    const val INVALID_CONFIGURATION = "INVALID_CONFIGURATION"
    const val TARGET_UNREACHABLE = "TARGET_UNREACHABLE"
    const val SESSION_ALREADY_RUNNING = "SESSION_ALREADY_RUNNING"
}

File: packages/network_analyzer_ios/ios/network_analyzer_ios/Sources/network_analyzer_ios/Monitoring/MonitorErrors.swift
import Foundation

/// The error codes the native side raises across the channel.
///
/// The Dart platform implementation maps each to a typed `Failure`, so no
/// `PlatformException` ever reaches a host application (constitution,
/// Principle III). Kept in one place because both platforms must agree.
enum MonitorErrors {
  static let permissionDenied = "PERMISSION_DENIED"
  static let unsupportedCapability = "UNSUPPORTED_CAPABILITY"
  static let gatewayDiscoveryFailed = "GATEWAY_DISCOVERY_FAILED"
  static let invalidConfiguration = "INVALID_CONFIGURATION"
  static let targetUnreachable = "TARGET_UNREACHABLE"
  static let sessionAlreadyRunning = "SESSION_ALREADY_RUNNING"
}
